src/auth.ts

```diff
@@ -34,44 +45,32 @@ const oauth2ClientSettings = {
   clientSecret: 'v6V3fKV_zWU7iw1DrpO1rknX',
   redirectUri: 'http://localhost',
 };
-const oauth2Client = new OAuth2Client(oauth2ClientSettings);
 
-// Google API clients
-export const script = google.script({version: 'v1', auth: oauth2Client}) as Script;
-export const logger = google.logging({version: 'v2', auth: oauth2Client}) as Logging;
-export const drive = google.drive({version: 'v3', auth: oauth2Client}) as Drive;
+const globalOauth2Client = new OAuth2Client(oauth2ClientSettings);
+
+// *Global* Google API clients
+export const script = google.script({version: 'v1', auth: globalOauth2Client}) as Script;
+export const logger = google.logging({version: 'v2', auth: globalOauth2Client}) as Logging;
+export const drive = google.drive({version: 'v3', auth: globalOauth2Client}) as Drive;
 export const discovery = google.discovery({version: 'v1'}) as Discovery;
 
 /**
  * Requests authorization to manage Apps Script projects.
  * @param {boolean} useLocalhost True if a local HTTP server should be run
  *     to handle the auth response. False if manual entry used.
- * @param {string} creds location of credentials file.
+ * @param {boolean} ownCreds save local rc file.
+ * @param {Array<string>} [scopes=[]] authorize additional OAuth scopes.
  */
-async function authorize(useLocalhost: boolean, creds: string) {
-  let ownCreds = false;
-  try {
-    const credentials = JSON.parse(fs.readFileSync(creds, 'utf8'));
-    if (credentials && credentials.installed && credentials.installed.client_id
-      && credentials.installed.client_secret) {
-        oauth2ClientSettings.clientId = credentials.installed.client_id;
-        oauth2ClientSettings.clientSecret = credentials.installed.client_secret;
-        ownCreds = true;
-        console.log(LOG.CREDENTIALS_FOUND);
-    } else {
-      logError(null, ERROR.BAD_CREDENTIALS_FILE);
-    }
-  } catch(err) {
-    if (err.code === 'ENOENT') {
-      logError(null, ERROR.NO_CREDENTIALS);
-    }
-    console.log(LOG.DEFAULT_CREDENTIALS);
-  }
+export async function authorize(useLocalhost: boolean, ownCreds: boolean, scopes: string[] = []) {
```

Comment:
This method signature is getting a bit big. Can we say:

```ts
export async function authorize(options: {
  useLocalhost: boolean,
  ownCreds: boolean,
  scopes: string[],
}) {
  
}
```

Also, it doesn't make sense to auth with no scopes in this case.